app/api/route.ts

```diff
@@ -1,6 +1,339 @@
+          displayName: "NOGD Events",
+          self: true,
+        },
+        start: {
+          dateTime: "2023-07-18T19:00:00-05:00",
+          timeZone: "America/Chicago",
+        },
+        end: {
+          dateTime: "2023-07-18T21:00:00-05:00",
+          timeZone: "America/Chicago",
+        },
+        recurringEventId: "1t6ian0r9kg4v6ruoog37df6js_R20230315T000000",
+        originalStartTime: {
+          dateTime: "2023-07-18T19:00:00-05:00",
+          timeZone: "America/Chicago",
+        },
+        iCalUID: "1t6ian0r9kg4v6ruoog37df6js_R20230315T000000@google.com",
+        sequence: 0,
+        extendedProperties: {
+          private: {
+            "X-MOZ-LASTACK": "20230815T225900Z",
+          },
+        },
+        eventType: "default",
+      },
+      {
+        kind: "calendar#event",
+        etag: '"3383036282414000"',
+        id: "nmmui7btr2h5uqpgpm030tjnkp_20230720T003000Z",
+        status: "confirmed",
+        htmlLink:
+          "https://www.google.com/calendar/event?eid=bm1tdWk3YnRyMmg1dXFwZ3BtMDMwdGpua3BfMjAyMzA3MjBUMDAzMDAwWiBjXzgxN3BzcDVlbWxnaDh2Z2c0MzJ1ZHJ0djIwQGc",
+        created: "2022-11-06T17:37:22.000Z",
+        updated: "2023-08-08T18:09:01.207Z",
+        summary: "SHO' & TELL",
+        description:
+          "It is time to Show and Tell!, hosted tonight by @thechamber69 grab the hdmi cable and show off what you've been up to! What you're working on isn't digital? WHO CARES, come by and show it off anyways!\u003cbr /\u003e\u003cbr /\u003e\u003cbr /\u003e\u003cbr /\u003eStarts at 7:30pm, hosted at Sea Cave, 3000 St. Claude.",
+        location: "Sea Cave, 3000 St Claude Ave, New Orleans, LA 70117, USA",
+        creator: {
+          email: "a@neworleansgamedev.com",
+        },
+        organizer: {
+          email: "c_817psp5emlgh8vgg432udrtv20@group.calendar.google.com",
+          displayName: "NOGD Events",
+          self: true,
+        },
+        start: {
+          dateTime: "2023-07-19T19:30:00-05:00",
+          timeZone: "America/Chicago",
+        },
+        end: {
+          dateTime: "2023-07-19T21:00:00-05:00",
+          timeZone: "America/Chicago",
+        },
+        recurringEventId: "nmmui7btr2h5uqpgpm030tjnkp",
+        originalStartTime: {
+          dateTime: "2023-07-19T19:30:00-05:00",
+          timeZone: "America/Chicago",
+        },
+        iCalUID: "nmmui7btr2h5uqpgpm030tjnkp@google.com",
+        sequence: 3,
+        extendedProperties: {
+          private: {
+            "X-MOZ-LASTACK": "20230816T232900Z",
+          },
+        },
+        eventType: "default",
+      },
+    ],
+  },
+];
+
+let cal: calendar_v3.Calendar;
+
+if (process.env.NODE_ENV == "production") {
+  cal = new calendar_v3.Calendar({ auth: process.env.CALENDAR_KEY });
+}
+
+// Get the latest week
+export async function GET(_: Request) {
+  if (process.env.NODE_ENV == "development") {
+    const dummyEvents: Event[] = dummyJSON.flatMap((item) =>
```

Comment:
flatMap with a map inside of it? ill have to pull this down and take a better look at it asap. im out of town and need to set up a chromebook w linux to get my IDE goin.

but, this seems like there is probably a simpler solution with better time complexity

Reply:
We're iterating over a data-structure of lists inside lists. I went a little crazy with the functional iterators (spent more time than I'd like to admit).

IIRC, I think the main thing is that it reallocates arrays (functional programming languages move by copying), and it might also be a little more readable if it was changed to imperative iterators (plus doesn't automatically allocate).

Something like this might work better? (mock-up):
```ts
let mockEvents: Event[] = [];
for (const eventList of mockEventJSON) {
  for (const event of eventList.items) {
    mockEvents.push({
      htmlLink: event.htmlLink,
      summary: event.summary,
      description: event.description,
      location: event.location,
      start: {
        dateTime: event.start.dateTime,
      },
      end: {
        dateTime: event.end.dateTime,
      },
    })
  }
}
```

Time complexity still ain't great though :/.

Reply:
okay i see, yeah i should have examined the data more closely. sometimes you just have to have nested iterators. i think either implementation is fine. the nested for ofs are probably slightly easier to read. thank you for taking the extra time to work out two working solutions.